From 29b4f7637227d4169e5b4a57daf363351e45afd0 Mon Sep 17 00:00:00 2001
From: Liu Ying <b17645@freescale.com>
Date: Mon, 2 Feb 2009 13:48:52 +0800
Subject: [PATCH] ENGR00106773 IPUv3:workaround full screen issue when previewing captured video

After previewing captured video on foreground with nonzero window offset,
full screen previewing on foreground fails. Workaround by setting window
offset to zero after finishing previewing.

Signed-off-by: Liu Ying <b17645@freescale.com>
---
 drivers/media/video/mxc/capture/ipu_prp_vf_sdc.c |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/drivers/media/video/mxc/capture/ipu_prp_vf_sdc.c b/drivers/media/video/mxc/capture/ipu_prp_vf_sdc.c
index dda0d65..6c67bfc 100644
--- a/drivers/media/video/mxc/capture/ipu_prp_vf_sdc.c
+++ b/drivers/media/video/mxc/capture/ipu_prp_vf_sdc.c
@@ -1,5 +1,5 @@
 /*
- * Copyright 2004-2008 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright 2004-2009 Freescale Semiconductor, Inc. All Rights Reserved.
  */
 
 /*
@@ -345,6 +345,8 @@ static int prpvf_stop(void *private)
 	if (cam->overlay_active == false)
 		return 0;
 
+	ipu_disp_set_window_pos(MEM_FG_SYNC, 0, 0);
+
 	if (cam->rotation >= IPU_ROTATE_90_RIGHT) {
 		ipu_unlink_channels(CSI_PRP_VF_MEM, MEM_ROT_VF_MEM);
 		ipu_unlink_channels(MEM_ROT_VF_MEM, MEM_FG_SYNC);
-- 
1.5.4.4

